core/network/helpers/compressor/zstd_stream_compressor.cpp

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <memory>
+#include <exception>
+
+#include <zstd.h>
+
+#include "zstd_stream_compressor.h"
+
+#include "zstd_error.h"
+
+namespace kagome::network {
+outcome::result<std::vector<uint8_t>> ZstdStreamCompressor::compress(std::span<uint8_t> data) try {
```

Comment:
Why not use `ZSTD_compress`/`ZSTD_decompress` like [uncompressCodeIfNeeded](https://github.com/qdrvm/kagome/blob/565921f314517b1877cf782bec715a8568eb3ac3/core/runtime/common/uncompress_code_if_needed.cpp#L46)?

Reply:
Because it won't work, I tried it. The sream compression is used here, you can find the reference to rust code in RFC

Reply:
I tested it directly with polkadot instance running on this commit, which is mentioned in FRC https://github.com/liuchengxu/polkadot-sdk/commit/2556fefacd2e817111d838af5f46d3dfa495852d#diff-d9656480fbba5813d9d8ff38b5f0661a6019d0b793024a751a1140034fc32747R268

Reply:
[rust-cpp-zstd.zip](https://github.com/user-attachments/files/16833718/rust-zstd.zip)